packages/react-component-library/src/components/DatePicker/DatePicker.tsx

```diff
@@ -154,6 +154,25 @@ export interface DatePickerProps
   jumpToToday?: boolean
 }
 
+const isDateDisabled = (
+  date: Date,
+  disabledDays?: DayPickerProps['disabled']
+) => {
+  if (!disabledDays) {
+    return false
+  }
+
+  if (typeof disabledDays === 'function') {
+    return disabledDays(date)
+  }
```

Comment:
`Matcher` already supports functions (and the `disabled` prop is typed as `Matcher | Matcher[] | undefined;`). 

E.g.:

```ts
> isMatch(new Date(), [() => true])
true
> isMatch(new Date(), [() => false])
false
```

Reply:
Oops yeah you're right my mistake.

Reply:
Looks like there was already an implementation of this elsewhere: `isDateInMatcher`.

Refactored to utilise that.